### PR TITLE
Add HTTPDriver interface and chromedp implementation

### PR DESCRIPTION
**Pull request by markwinter**
I rebased the previous PR as it had gotten stale https://github.com/gocolly/colly/pull/503 and I think several people are eager to use colly with chromedp

Will look to add tests and fix any builds. Can also squash commits when ready

### CODE REVIEW

File: colly.go [hunk 1]
```diff
@@ -485,27 +490,6 @@ func (c *Collector) Init() {
...
-// Google App Engine. Example:
-//
-//	func startScraper(w http.ResponseWriter, r *http.Request) {
-//	  ctx := appengine.NewContext(r)
-//	  c := colly.NewCollector()
-//	  c.Appengine(ctx)
-//	   ...
-//	  c.Visit("https://google.ca")
-//	}
-func (c *Collector) Appengine(ctx context.Context) {
```
Comment (markwinter) [inline colly.go:500]: Should use the functinal options instead

```go
colly.NewCollector(
  SetHTTPDriver(NewAppEngineBackend(ctx))
)
```

**Comment by marcalc** [comment]
@markwinter Why was your PR closed? Was it merged somewhere else? Appreciate it.

**Comment by markwinter** [comment]
@marcalc Just because I didn't think it would be merged and didn't want to keep it up to date

**Comment by asciimoo** [comment]
@markwinter This would be a very useful addition & the PR is clean. We are trying to find maintainers to resurrect the project and keep it well maintained. I'd appreciate a lot if you could rebase it one last time.

Sorry for the long delayed response.

**Comment by markwinter** [comment]
@asciimoo Thanks for the feedback, I will rebase and open the PR again

**Comment by markwinter** [comment]
Opened a new MR here https://github.com/gocolly/colly/pull/844

**Pull request closed without merging**